python/ray/_private/utils.py

```diff
@@ -1258,7 +1258,13 @@ def init_grpc_channel(
             private_key=private_key,
             root_certificates=ca_cert,
         )
-        channel = grpc_module.secure_channel(address, credentials, options=options)
+        
+        try:
+            metadata = load_metadata_from_env()
```

Comment:
based on the ray client implementation, the metadata needs to be passed to the stub method call? Does that mean we need to find all the stub calls and add metadata?

Reply:
Well, only the stubs that are utilized by the CLI when accessing the Ray Cluster. This was implemented with ray.init() meaning users can pass `ray.init(_metadata={})`.

Reply:
So what's the next step of this PR? Find all the related stubs and add metadata?

Also I realized that some CLIs (e.g. job submission CLI) are talking to the cluster through HTTP instead of GRPC, we should also think about that as well (but I think it's out of scope of this PR).

Reply:
I think I could use some guidance. Unsure which CLI commands would be best to support this feature, and areas at code to start looking into.

Reply:
For example, which CLI commands should support metadata? Is there a best format for adding new params for cli commands? Pointing to code files would be helpful, just want to make sure we implement this correctly. Thank you!!!

Reply:
Hi @jjyao, wanted to follow up on this discussion. Thanks!

Reply:
Sorry for the late reply. I feel the most important CLIs to support are actually `ray job submit` (https://github.com/ray-project/ray/blob/75b206e419f88d25e5e08777b2548600287e2120/dashboard/modules/job/cli.py#L158), `ray logs`, `ray list`  which are based on https instead of grpc. But I think the same principle applies. Do you mind if we start with those?

Reply:
Or we can do `ray status` which is based on grpc: https://github.com/ray-project/ray/blob/75b206e419f88d25e5e08777b2548600287e2120/python/ray/scripts/scripts.py#L1872

Reply:
I like the path of going both ways. I'm curious if there is an init command that will persist this metadata or cookies that are passed to all calls?

For example, could we use an environment variable?

Reply:
The JobSubmissionClient supports this already, if I am reading the code correctly: https://github.com/ray-project/ray/blob/75b206e419f88d25e5e08777b2548600287e2120/dashboard/modules/dashboard_sdk.py#L125

Can you take a quick poke and verify? Thanks so much for the collaboration!

Reply:
This metadata is not for authentication but it also supports http headers that can be used for authentication. Seems you can pass in headers if you are using job submission apis programatically. But currently you cannot pass those using `ray job submit` CLI. @architkulkarni is my understanding correct? Is this something we want to support in CLI as well. I think @peterghaddad is willing to contribute here.

Reply:
It would be great to support these in the Ray Jobs CLI! I think it just hasn't been prioritized yet.

Reply:
Happy to work these. Do you think an environment variable would be sufficient? This may be the way to go so that the end-user doesn't have to type in authentication metadata per cli command?

Reply:
Oh @peterghaddad I just realized I was mistaken-- in the Jobs API, the `metadata` field is used for arbitrary user-provided metadata which is eventually just internally stored at https://github.com/ray-project/ray/blob/master/python/ray/job_config.py#L29, so I don't think it works for the kind of authentication uses you have in mind (it never gets read by anything).  

Can what you're doing be accomplished just using the `headers` option?  If this is missing from the jobs CLI, it would be good to add it there.  Having an option to specify it via an environment variable makes sense to me.

Reply:
Believe so, so the `headers` option is what's passed through network traffic? i.e cookies etc.

Reply:
Yes.